core/deliverservice/deliveryclient.go

```diff
@@ -279,7 +279,7 @@ func (d *deliverServiceImpl) newClient(chainID string, ledgerInfoProvider blocks
 		}
 		sleepIncrement := float64(time.Millisecond * 500)
 		attempt := float64(attemptNum)
-		return time.Duration(math.Min(math.Pow(2, attempt)*sleepIncrement, reconnectBackoffThreshold)), true
+		return time.Duration(math.Min(math.Pow(2, attempt)*sleepIncrement, reconnectBackoffThreshold.Hours())), true
```

Comment:
I don't think you want to return `Hours()` here since your sleep increment is just casting the duration to float64. The threshold should match the base unit used for the increment. Probably just casting to a float64 is enough.